src/main/java/gregtech/api/metatileentity/SimpleMachineMetaTileEntity.java

```diff
@@ -96,6 +97,18 @@ public boolean onWrenchClick(EntityPlayer playerIn, EnumHand hand, EnumFacing fa
         return super.onWrenchClick(playerIn, hand, facing, hitResult);
     }
 
+    @Override
+    public boolean placeCoverOnSide(EnumFacing side, ItemStack itemStack, CoverDefinition coverDefinition) {
+        boolean coverPlaced = super.placeCoverOnSide(side,itemStack,coverDefinition);
+        if (getOutputFacing() == side && !isAllowInputFromOutputSide()) {
+            if (getCoverAtSide(side).getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY,null) != null ||
+                (getCoverAtSide(side).getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, null) != null)) {
```

Comment:
I still think this is overkill and I would rather see it as check for existence of Interface or even call function on CoverBehavior which will return info if this cover should setAllowInputFromOutputSide.

Or is there something I am not seeing and you have to do it this way?

Reply:
still learning how java actually works, i guess. i was under the impression i needed to do it like that because only that class has access to return the actual output side.
Anyway figured out how to move that function to the cover behavior API, that had a onAttached hook, so the code would only run if the cover was actually placed. Since only conveyors/arms/pumps can move things by themselves, it doesnt make sense other covers messing with the "allowInputFromOutputSide"
Thanks for the patience.

Reply:
Sure no problem about taking time to learn or work on it as long you properly communicate.
I see what you did and kind of understand why you did it but it is not correct. Fundamentally Covers were design in way that they don't need to know on what they are placed on, that is handled by `ICoverable`. By casting `coverHolder` to `SimpleMachineMetaTileEntity` you made coupling which goes against design (anti-pattern). So this has to be done other way around as Machines are aware of `CoverBehavior` and other high level Cover classes (not Covers by themselves). Which means that easiest and best solution is in `CoverBehavior` create function something like `canOverrideAllowInputFromOutputSide` which will return `false`. All Covers which support this behavior will override this function and return `true`. And then in `placeCoverOnSide` in `SimpleMachineMetaTileEntity` you will check it and set `allowInputFromOutputSide`.